flips/20211007-transaction-fees.md

```diff
@@ -0,0 +1,327 @@
+
+[^1]: Assuming there weren't any changes to the state that would make the transactions script execute differently.
+
+Finally we need to multiply the transaction fees by a (unit-less) surge factor that would be more than 1 in the case the network is experiencing heavy load.
+
+### Terminology
+
+- <img src="https://render.githubusercontent.com/render/math?math=F"> : Transaction fees.
+- <img src="https://render.githubusercontent.com/render/math?math=s"> : Surge factor; unit-less factor that is higher if the network is experiencing heavy load.
+- <img src="https://render.githubusercontent.com/render/math?math=F_I"> : Inclusion part of the transaction fees.
+- <img src="https://render.githubusercontent.com/render/math?math=F_E"> : Execution part of the transaction fees.
+- <img src="https://render.githubusercontent.com/render/math?math=I">: Inclusion effort; The time invariant effort required to handle this transaction (include it into a collection, check signatures are valid, check that the payer can pay, transfer the transaction over the wire,...). The inclusion effort is known before the transaction is executed.
+- <img src="https://render.githubusercontent.com/render/math?math=E">: Execution effort; The time invariant effort required to execute this transaction (run the script, process the results, send results to verification node, verify the results, ...). The Execution effort cannot be know without executing the 
+- <img src="https://render.githubusercontent.com/render/math?math=E%5E%7B%5Ctext%7Bmax%7D%7D">: Maximum execution effort (gas/computation limit).
+- <img src="https://render.githubusercontent.com/render/math?math=p_E(E)"> or <img src="https://render.githubusercontent.com/render/math?math=p_I(I)">: Flow cost of effort. In general execution and inclusion effort can be priced differently.
+
+The transaction fees equation can then be written in the following form.
+
+<img src="https://render.githubusercontent.com/render/math?math=F%20%3D%20s%20%5Cleft%5B%20F_I%20%2B%20F_E%20%5Cright%5D%20%3D%20s%20%5Cleft%5B%20p_I(I)%20%2B%20p_E(E)%20%5Cright%5D">
+
+The maximum fees a transaction can incur can be written with the equation below. 
+
+<img src="https://render.githubusercontent.com/render/math?math=F%5E%7B%5Ctext%7Bmax%7D%7D%20%3D%20s%20%5Cleft%5B%20p_I(I)%20%2B%20p_E(E%5E%7B%5Ctext%7Bmax%7D%7D)%20%5Cright%5D">
+
+### Surge factor
+
+The surge factor serves only one purpose, which is to give all transactions on the network a surcharge or potentially a discount when the network load is high or low respectively. It should be defined on a smart contract and adjustable via the service account admin resource and should be accessible for anyone to read. 
+
+The surge factor could be adjusted manually from the start, by manually moniyoring the network and sending a transaction to update the surge factor. This would mean responding to network load would be very slow. 
```

Comment:
```suggestion
The surge factor could be adjusted manually from the start, by manually monitoring the network and sending a transaction to update the surge factor. This would mean responding to network load would be very slow. 
```